### PR TITLE
[ticket/17151] Make macros available for extensions

### PR DESCRIPTION
**Pull request by rxu**
Also add tests and error reporting.

PHPBB3-17151

Checklist:

- [x] Correct branch: master for new features; 3.3.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master](https://area51.phpbb.com/docs/master/coding-guidelines.html) and [3.3.x](https://area51.phpbb.com/docs/dev/3.3.x/development/coding_guidelines.html)
- [x] Commit follows commit message [format](https://area51.phpbb.com/docs/dev/3.3.x/development/git.html)

Tracker ticket:

https://tracker.phpbb.com/browse/PHPBB3-17151

### CODE REVIEW

**Comment by rxu** [comment]
Also currently forms macros is not used for building `select`s, but that can be resolved later in a separate PR.

File: tests/functional/fixtures/ext/foo/bar/styles/prosilver/template/foobar_mcp.html [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<!-- INCLUDE overall_header.html -->
```
Comment (marc1706) [inline tests/functional/fixtures/ext/foo/bar/styles/prosilver/template/foobar_mcp.html:1]: Should use twig syntax instead

File: tests/functional/fixtures/ext/foo/bar/mcp/main_info.php [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+*
+*/
+
+namespace foo\bar\mcp;
+
+class main_info
+{
+	function module()
+	{
+		return array(
```
Comment (marc1706) [inline tests/functional/fixtures/ext/foo/bar/mcp/main_info.php:20]: Short array syntax in this file

File: tests/functional/fixtures/ext/foo/bar/ucp/main_info.php [hunk 3]
```diff
@@ -19,9 +19,9 @@ function module()
 	{
 		return array(
```
Comment (marc1706) [inline tests/functional/fixtures/ext/foo/bar/ucp/main_info.php:20]: Would also opt for changing this file to short array syntax while you're changing it

**Review by marc1706 — CHANGES_REQUESTED** [review]

**Review by marc1706 — APPROVED** [review]

**Pull request merged**